src/libcore/num/f32.rs

```diff
@@ -400,6 +402,31 @@ impl f32 {
         intrinsics::minnumf32(self, other)
     }
 
+    /// Rounds toward zero and converts to any primitive integer type,
+    /// assuming that the value is finite and fits in that type.
+    ///
+    /// ```
+    /// #![feature(float_round_unchecked_to)]
+    ///
+    /// let value = 4.6_f32;
+    /// let rounded = unsafe { value.round_unchecked_to::<u16>() };
+    /// assert_eq!(rounded, 4);
+    ///
+    /// let value = -128.9_f32;
+    /// let rounded = unsafe { value.round_unchecked_to::<i8>() };
+    /// assert_eq!(rounded, std::i8::MIN);
+    /// ```
+    ///
+    /// # Safety
+    ///
+    /// The value must be finite and fit in the return type.
```

Comment:
Maybe this could be elaborated upon what finite and fitting entails?

Reply:
Finite is defined in the docs of the `is_finite` method, but I can repeat in this comment.

As discussed in the PR description, “fit” is copied from https://llvm.org/docs/LangRef.html#fptoui-to-instruction and I couldn’t anything find more precise. I’d prefer to find out what LLVM does exactly rather than documenting suppositions as fact, but I’m not sure how.

Reply:
Re. finite, maybe link to `is_finite` saying "as defined by `is_finite`"?

Hopefully @rkruppe is more knowledgeable re. "fit".

Reply:
This bit of doc-comment is now “The value must be finite (not infinite or `NaN`) and fit in the return type.” and I’ve added a (non-doc-) comment with a link to relevant LLVM docs.

Reply:
If it's not spelled out in LLVM's LangRef, then my working knowledge of how the instruction works probably isn't worth much. There's dozens of places in LLVM that could deviate from the expected semantics (various IR passes and utilities, as well as the lowerings in the backends) and I do not know all of them.